pkg/gofr/http/middleware/oauth.go

```diff
@@ -114,35 +183,50 @@ type PublicKeyProvider interface {
 }
 
 // OAuth is a middleware function that validates JWT access tokens using a provided PublicKeyProvider.
-func OAuth(key PublicKeyProvider) func(inner http.Handler) http.Handler {
+func OAuth(key PublicKeyProvider, config *ClaimConfig) func(http.Handler) http.Handler {
```

Comment:
I'm unsure it's the right direction.

First, it's a breaking change.

You could have used an Option pattern here too

Then you create a claim and you apply its options.

```suggestion
func OAuth(key PublicKeyProvider, opts ClaimOption...) func(http.Handler) http.Handler {
	config := &ClaimConfig{} 
	for _, opt := range opts {
		opt(config) 
	}
```


So the signature is retro-compatible and, you don't have to do most of the changes you made in oauth_test.go file

Reply:
Wouldn't this get repetitive ? As I am already doing this for EnableOAuth in auth.go ..... , line no. 87.

Reply:
I hope this will be clearer

https://github.com/gofr-dev/gofr/pull/1580#discussion_r2001021304

Reply:
Yes, the other comment makes it clearer.